packages/flutterfire_ui/lib/src/auth/email_verification.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+import 'package:flutter/material.dart';
+
+enum EmailVerificationState {
+  unresolved,
+  unverified,
+  dismissed,
+  sending,
+  pending,
+  sent,
+  verified,
+  failed,
+}
+
+class EmailVerificationService extends ValueNotifier<EmailVerificationState> {
+  final FirebaseAuth auth;
+
+  EmailVerificationService(this.auth)
+      : super(EmailVerificationState.unresolved);
+
+  User get user => auth.currentUser!;
+  EmailVerificationState get state => value;
+
+  Exception? error;
+
+  bool _isMobile(TargetPlatform platform) {
+    return platform == TargetPlatform.android || platform == TargetPlatform.iOS;
+  }
+
+  Future<void> reload() async {
+    await user.reload();
+    if (user.emailVerified) {
+      value = EmailVerificationState.verified;
+    } else {
+      value = EmailVerificationState.unverified;
+    }
+  }
+
+  void dismiss() {
+    value = EmailVerificationState.dismissed;
+  }
+
+  Future<void> sendVerificationEmail(
+    TargetPlatform platform,
+    ActionCodeSettings? actionCodeSettings,
+  ) async {
+    value = EmailVerificationState.sending;
+    try {
+      await user.sendEmailVerification(actionCodeSettings);
+    } on Exception catch (e) {
+      error = e;
```

Comment:
Is there a worry this error could be smothered if there is an error in the next try/catch?

Reply:
oh, actually if there is an exception in this phase there is no point in proceeding